Sources/Source/Components/Buttons/ButtonStyle+SourceButtonStyle.swift

```diff
@@ -0,0 +1,174 @@
+import SwiftUI
+import GuardianFonts
+
+struct SourceButtonStyle: ButtonStyle {
+    let buttonSize: ButtonSize
+    let buttonPriority: ButtonPriority
+    let buttonTheme: ButtonTheme
+
+    @Environment(\.colorScheme) private var colorScheme
+
+    private var disabledOpacity: CGFloat {
+        return colorScheme == .dark ? 0.6 : 0.4
+    }
+
+    func makeBody(configuration: Configuration) -> some View {
+        configuration.label
+            .frame(maxWidth: .infinity)
+            .padding(.vertical, buttonSize.verticalPad)
+            .padding(.horizontal, buttonSize.horizontalPad)
+            .font(GuardianFont(style: .textSansBold, size: buttonSize.fontSize))
+            .foregroundColor(foregroundColor(for: buttonPriority))
+            .background(backgroundShape(for: buttonPriority))
+            .opacity(configuration.isPressed ? 0.8 : 1)
+            .scaleEffect(configuration.isPressed ? 0.95 : 1)
+            .lineLimit(1)
+    }
+
+    private func foregroundColor(for priority: ButtonPriority) -> Color {
+        switch priority {
+        case .primary:
+            return Color(uiColor: buttonTheme.foregroundColorPrimary)
+        case .secondary:
+            return Color(uiColor: buttonTheme.foregroundColorSecondary)
+        case .tertiary:
+            return Color(uiColor: buttonTheme.foregroundColorTertiary)
+        case .subdubed:
+            return Color(uiColor: buttonTheme.foregroundColorSubdued)
+        }
+    }
+
+    private func backgroundShape(for priority: ButtonPriority) -> some View {
+        Group {
+            switch priority {
+            case .primary:
+                Capsule()
+                    .fill(Color(uiColor: buttonTheme.backgroundColorPrimary))
+            case .secondary:
+                Capsule()
+                    .fill(Color(uiColor: buttonTheme.backgroundColorSecondary))
+            case .tertiary:
+                Capsule()
+                    .stroke(Color(uiColor: buttonTheme.foregroundColorTertiary))
+            case .subdubed:
+                EmptyView()
+
+            }
+        }
+    }
+}
+
+extension ButtonStyle where Self == SourceButtonStyle {
+    static func source(size: ButtonSize, priority: ButtonPriority, theme: ButtonTheme) -> SourceButtonStyle {
+        SourceButtonStyle(buttonSize: size, buttonPriority: priority, buttonTheme: theme)
+    }
```

Comment:
This will need to be public so that is accessible in other projects.

Reply:
Good spot! Thanks. Have made that adjustment. Had to make a couple other things public too as a result. Still waiting for sign off from design, so will hold off merging until I've got the all clear from them.